File: 7-kyu/reverse-a-number/c++/solution.cpp
#include <cmath>

long long reverseNumber(long long n)
{
    if (n > -10 && n < 10) return n;
    bool less_than_zero = n < 0;
    n = std::llabs(n);
    long long ret = 0;
    long long mult = std::floor(std::log10(n));
    while (n > 0) {
        ret += (n % 10) * std::pow(10, mult);
        mult--;
        n /= 10;
    }
    return ret * (less_than_zero ? -1 : 1);
}

File: 7-kyu/reverse-a-number/python/solution.py
from math import floor, log10

def reverse_number(n):
    if n > -10 and n < 10:
        return n
    n_abs = abs(n)
    ret = 0
    mult = floor(log10(n_abs))
    while n_abs > 0:
        ret += (n_abs % 10) * (10 ** mult)
        mult -= 1
        n_abs //= 10
    return ret * (-1 if n < 0 else 1)